dfastmi/batch/ConfigurationCheckerLegacy.py

```diff
@@ -0,0 +1,426 @@
+        """
+        Check if a version 1 analysis configuration is valid.
+
+        Arguments
+        ---------
+        rivers: RiversObject
+            A dictionary containing the river data.
+        config : configparser.ConfigParser
+            Configuration for the D-FAST Morphological Impact analysis.
+
+        Returns
+        -------
+        success : bool
+            Boolean indicating whether the D-FAST MI analysis configuration is valid.
+        """
+        try:
+            reach = self._get_reach(rivers, config)
+        except LookupError:
+            return False
+
+        nwidth = reach.normal_width
+        [_, apply_q, _, _, _, _, _, _] = self._get_levels(reach, config, nwidth)
+
+        mode_str = config.get("General", "Mode", fallback=DFLOWFM_MAP)
+        for i in range(3):
+            if not apply_q[i]:
+                continue
+
+            if not self._validate(mode_str, config, i) :
+                return False
+
+        return True
+
+    def _get_levels(
+        self,
+        reach: ReachLegacy,
+        config: configparser.ConfigParser,
+        nwidth: float
+    ) -> Tuple[Vector, BoolVector, float, Vector, float, Vector, Vector, Vector]:
+        """
+        Determine discharges, times, etc. for version 1 analysis
+
+        Arguments
+        ---------
+        rivers : RiversObject
+            A dictionary containing the river data.
+        ibranch : int
+            Number of selected branch.
+        ireach : int
+            Number of selected reach.
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        nwidth : float
+            Normal river width (from rivers configuration file) [m].
+
+        Return
+        ------
+        discharges : Vector
+            Array of discharges (Q); one for each forcing condition [m3/s].
+        apply_q : BoolVector
+            A list of flags indicating whether the corresponding entry in Q should be used.
+        q_threshold : float
+            River discharge at which the measure becomes active [m3/s].
+        time_mi : Vector
+            A vector of values each representing the fraction of the year during which the discharge Q results in morphological impact [-].
+        tstag : float
+            Fraction of year during which flow velocity is considered negligible [-].
+        fractions_of_the_year : Vector
+            A vector of values each representing the fraction of the year (T) during which the discharge is given by the corresponding entry in Q [-].
+        rsigma : Vector
+            A vector of values each representing the relaxation factor for the period given by the corresponding entry in Q [-].
+        celerity : Vector
+            A vector of values each representing the bed celerity for the period given by the corresponding entry in Q [m/s].
+        """
+        q_stagnant = reach.qstagnant
```

Comment:
Is there a specific reaosn these are split of into seperate variables and then passed together with reach into the next method?

Reply:
discussed, no action